src/ranged.cpp

```diff
@@ -1042,21 +1042,55 @@ static int print_aim_bars( const player &p, WINDOW *w, int line_number, item *we
     // Fairly arbitrary cap on steadiness...
     const double steadiness = 1.0 - steady_score / MIN_RECOIL;
 
-    const std::array<std::pair<double, char>, 3> confidence_ratings = {{
-        std::make_pair( p.projectile_attack_chance( dispersion, range, accuracy_headshot ), '*' ),
-        std::make_pair( p.projectile_attack_chance( dispersion, range, accuracy_goodhit ), '+' ),
-        std::make_pair( p.projectile_attack_chance( dispersion, range, accuracy_grazing ), '|' ) }};
+    const auto get_chance = [&p, dispersion, range]( double acc ) {
+        return p.projectile_attack_chance( dispersion, range, acc );
+    };
+
+    // This could be extracted, to allow more/less verbose displays
+    using cconf = std::tuple<double, char, std::string>;
+    static const std::array<cconf, 3> confidence_config = {{
+        std::make_tuple( accuracy_headshot, '*', _( "Headshot" ) ),
+        std::make_tuple( accuracy_goodhit, '+', _( "Hit" ) ),
+        std::make_tuple( accuracy_grazing, '|', _( "Graze" ) )
+    }};
 
     const int window_width = getmaxx( w ) - 2; // Window width minus borders.
-    const std::string &confidence_bar = get_labeled_bar( 1.0, window_width, _( "Confidence" ),
-                                                         confidence_ratings.begin(),
-                                                         confidence_ratings.end() );
-    const std::string &steadiness_bar = get_labeled_bar( steadiness, window_width,
-                                                         _( "Steadiness" ), '*' );
 
-    mvwprintw( w, line_number++, 1, _( "Symbols: * = Headshot + = Hit | = Graze" ) );
-    mvwprintw( w, line_number++, 1, confidence_bar.c_str() );
-    mvwprintw( w, line_number++, 1, steadiness_bar.c_str() );
+    if( get_option<std::string>( "ACCURACY_DISPLAY" ) == "numbers" ) {
+        // Chances by get_chance are absolute, meaning graze will always be higher than headshot etc.
+        int last_chance = 0;
+        std::string confidence_s = enumerate_as_string( confidence_config.begin(), confidence_config.end(),
+        [&]( const cconf &config ) {
+            // @todo Consider not printing 0 chances, but only if you can print something (at least miss 100% or so)
+            int chance = std::min<int>( 100, 100.0 * get_chance( std::get<0>( config ) ) ) - last_chance;
+            last_chance += chance;
+            return string_format( "%s: %3d%%", std::get<2>( config ).c_str(), chance );
+        }, false );
+        line_number += fold_and_print_from( w, line_number, 1, window_width, 0,
+                                            c_white, confidence_s );
+
+        std::string steadiness_s = string_format( "%s: %d%%", _( "Steadiness" ), (int)( 100.0 * steadiness ) );
+        mvwprintw( w, line_number++, 1, steadiness_s.c_str() );
```

Comment:
This line causes "Bad format string for printf" (because of the percent sign in it). It has to be rewritten: 
```c++
mvwprintw( w, line_number++, 1, "%s", steadiness_s.c_str() );
```